### PR TITLE
platform: make IOMUX, SPI, DW_GPIO a Kconfig option

### PR DESCRIPTION
**Pull request by bkokoszx**
This commit:
1. Makes IOMUX, SPI and DW_GPIO as a configurable
   Kconfig options.
2. Replace CONFIG_{PLATFORM} defines by CONFIG_CAVS_VERSION
   in cavs/platform.c source file.

Signed-off-by: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>

### CODE REVIEW

File: src/platform/intel/cavs/platform.c [hunk 1]
```diff
@@ -173,8 +174,7 @@ struct work_queue_timesource platform_generic_queue[] = {
 	.timer_clear	= platform_timer_clear,
 	.timer_get	= platform_timer_get,
 },
-#if defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE) \
-	|| defined(CONFIG_SUECREEK)
+#if CAVS_VERSION >= CAVS_VERSION_1_8
```
Comment (mmaka1) [inline src/platform/intel/cavs/platform.c]: Seems to be independent change, good candidate for a dedicated commit.

File: src/platform/intel/cavs/platform.c [hunk 2]
```diff
@@ -312,32 +315,35 @@ static struct spi_platform_data spi = {
...
+int platform_boot_complete(uint32_t boot_message)
+{
 	mailbox_dspbox_write(0, &ready, sizeof(ready));
 #if defined(CONFIG_MEM_WND)
 	mailbox_dspbox_write(sizeof(ready), &sram_window,
 		sram_window.ext_hdr.hdr.size);
 #endif // defined(CONFIG_MEM_WND)
 
-#if defined(CONFIG_APOLLOLAKE)
+#if CAVS_VERSION == CAVS_VERSION_1_5
```
Comment (mmaka1) [inline src/platform/intel/cavs/platform.c]: Another one to be separated from new CONFIG_... options. In general, please move all CAVS_VERSION related changes to a single, separate commit.

File: Kconfig [hunk 3]
```diff
@@ -30,6 +30,18 @@ config MEM_WND
 	bool
 	default n
 
+config SPI
```
Comment (lgirdwood) [inline Kconfig]: irc, this is DW SPI IP ?

File: Kconfig [hunk 4]
```diff
@@ -30,6 +30,18 @@ config MEM_WND
...
+	bool
+	default n
+
+config IOMUX
```
Comment (lgirdwood) [inline Kconfig]: INTEL_IOMUX

File: src/platform/intel/cavs/platform.c (hunk 1 above)
Comment (bkokoszx) [reply src/platform/intel/cavs/platform.c]: I've moved CAVS_VERSION defines change to another commit

File: src/platform/intel/cavs/platform.c (hunk 2 above)
Comment (bkokoszx) [reply src/platform/intel/cavs/platform.c]: I've moved CAVS_VERSION defines change to another commit

File: Kconfig (hunk 3 above)
Comment (bkokoszx) [reply Kconfig]: Changed to DW_SPI

File: Kconfig (hunk 4 above)
Comment (bkokoszx) [reply Kconfig]: Changed to INTEL_IOMUX

**Pull request merged**